test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/RazorDirectivesTest/DirectiveDescriptor_FileScopedSinglyOccurring_ErrorsIfDuplicate.stree.txt

```diff
@@ -1,39 +1,46 @@
-            SyntaxKind.Dot;[.];
-            SyntaxKind.Identifier;[ASCIIEncoding];
-        Markup span - Gen<None> - [LF] - SpanEditHandler;Accepts:Whitespace - (42:0,42) - Tokens:1
-            SyntaxKind.NewLine;[LF];
-    Markup span - Gen<Markup> - [] - SpanEditHandler;Accepts:Any - (44:1,0) - Tokens:1
-        SyntaxKind.Unknown;[];
-    Directive block - Gen<Directive:{custom;SingleLine;FileScopedSinglyOccurring} [RZ2001(44:1,0 [7] )]> - 41 - (44:1,0)
-        Transition span - Gen<None> - [@] - SpanEditHandler;Accepts:None - (44:1,0) - Tokens:1
-            SyntaxKind.Transition;[@];
-        MetaCode span - Gen<None> - [custom] - SpanEditHandler;Accepts:None - (45:1,1) - Tokens:1
-            SyntaxKind.Identifier;[custom];
-        Code span - Gen<None> - [ ] - SpanEditHandler;Accepts:Whitespace - (51:1,7) - Tokens:1
-            SyntaxKind.Whitespace;[ ];
-        Code span - Gen<DirectiveToken {;Type;Opt:False}> - [System.Text.Encoding.UTF8Encoding] - DirectiveTokenEditHandler;Accepts:NonWhitespace - (52:1,8) - Tokens:7
-            SyntaxKind.Identifier;[System];
-            SyntaxKind.Dot;[.];
-            SyntaxKind.Identifier;[Text];
-            SyntaxKind.Dot;[.];
-            SyntaxKind.Identifier;[Encoding];
-            SyntaxKind.Dot;[.];
-            SyntaxKind.Identifier;[UTF8Encoding];
-    Markup span - Gen<Markup> - [] - SpanEditHandler;Accepts:Any - (85:1,41) - Tokens:1
-        SyntaxKind.Unknown;[];
+RazorDocument - [0..85)::85 - [@custom System.Text.Encoding.ASCIIEncodingLF@custom System.Text.Encoding.UTF8Encoding]
+    MarkupBlock - [0..85)::85
+        MarkupTextLiteral - [0..0)::0 - [] - Gen<Markup> - SpanEditHandler;Accepts:Any
+            Marker;[];
+        CSharpCodeBlock - [0..44)::44
+            RazorDirective - [0..44)::44
+                CSharpTransition - [0..1)::1 - Gen<None> - SpanEditHandler;Accepts:None
+                    Transition;[@];
+                RazorDirectiveBody - [1..44)::43
+                    RazorMetaCode - [1..7)::6 - Gen<None> - SpanEditHandler;Accepts:None
+                        Identifier;[custom];
+                    CSharpCodeBlock - [7..44)::37
+                        CSharpStatementLiteral - [7..8)::1 - [ ] - Gen<None> - SpanEditHandler;Accepts:Whitespace
+                            Whitespace;[ ];
+                        CSharpStatementLiteral - [8..42)::34 - [System.Text.Encoding.ASCIIEncoding] - Gen<DirectiveToken {;Type;Opt:False}> - DirectiveTokenEditHandler;Accepts:NonWhitespace
+                            Identifier;[System];
+                            Dot;[.];
+                            Identifier;[Text];
+                            Dot;[.];
+                            Identifier;[Encoding];
+                            Dot;[.];
+                            Identifier;[ASCIIEncoding];
+                        MarkupEphemeralTextLiteral - [42..44)::2 - [LF] - Gen<None> - SpanEditHandler;Accepts:Whitespace
```

Comment:
I see this and I lol? What is this?

Reply:
Naming is hard! I had originally named it `MarkupHiddenTextLiteral`. This is basically content that is necessary for design time but we don't want them rendered in output.

Reply:
It seems like I have a lot to learn